crates/bevy_pbr/src/render/mesh/mod.rs

```diff
@@ -1136,22 +1119,26 @@ impl<P: PhaseItem, const I: usize> RenderCommand<P> for SetMeshBindGroup<I> {
     fn render<'w>(
         _item: &P,
         _view: (),
-        (mesh_index, skinned_mesh_joints): ROQueryItem<'_, Self::ItemWorldQuery>,
-        mesh_bind_group: SystemParamItem<'w, '_, Self::Param>,
+        (mesh_index, skin_index): ROQueryItem<'_, Self::ItemWorldQuery>,
+        bind_groups: SystemParamItem<'w, '_, Self::Param>,
         pass: &mut TrackedRenderPass<'w>,
     ) -> RenderCommandResult {
-        if let Some(joints) = skinned_mesh_joints {
-            pass.set_bind_group(
-                I,
-                mesh_bind_group.into_inner().skinned.as_ref().unwrap(),
-                &[mesh_index.index(), joints.index],
+        let bind_groups = bind_groups.into_inner();
+        let has_skin = skin_index.is_some();
+        let Some(bind_group) = bind_groups.get(has_skin) else {
```

Comment:
We could probably use expect() and map() or something here, rather than an if let block.

Reply:
This is not a panic, it only logs an error and returns `Failure`. Note that I use a `let-else` here to avoid rightward drift.

Since we are returning from the `render` function in the `else` branch, it is impossible to use an Option combinator (such as map or expect) in this situation, since a closure cannot return from the encompassing function.

I extracted the error message to make the logic more apparent.